docs/source/x_ext.rst

```diff
@@ -531,13 +503,83 @@ The |processor| shall cause an illegal instruction fault when attempting to exec
+  |                        | <registerflags>`         | each register within the pair individually.                                                                     |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``ecs``                | logic [5:0]              | Extension Context Status ({``mstatus.xs``,``mstatus.fs``,``mstatus.vs``}).                                      |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``ecs_valid``          | logic                    | Validity of the Extension Context Status.                                                                       |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+
+There are two main scenarios, in how the register interface will be used:
+
+1. A register transaction can be started in the same clock cycle as the issue transaction (``issue_req_valid = register_valid`` and ``issue_req.id = register.id``).
+   In this case, the |processor| will speculatively provide all possible source registers via ``register.rs`` when they become available (signalled via the respective ``valid`` signals). 
+   The |coprocessor| will delay accepting the instruction until all necessary registers are provided, and only then assert ``issue_ready``.
+   The ``rs_valid`` bits are not required to be stable during the transaction.
+   Each bit can transition from 0 to 1, but is not allowed to transition back to 0 during a transaction.
+   A |coprocessor| is not expected to wait for all ``rs_valid`` bits to be 1, but only for those registers it intends to read.
+   The ``rs`` signals are only required to be stable during the part of a transaction in which these signals are considered to be valid.
+   The ``ecs_valid`` bit is not required to be stable during the transaction. It can transition from 0 to 1, but is not allowed to transition back to 0 during a transaction.
+   The ``ecs`` signal is only required to be stable during the part of a transaction in which this signals is considered to be valid.
+
+2. For a |processor| which splits the issue and register interface into subsequent pipeline stages (e.g. because it has a dedicated read registers (RR) stage), the registers will be provided after the issue transaction completed. 
+   The |processor| initiates the register transaction once all registers are available. 
+   If the |coprocessor| is able to accept multiple issue transactions before receiving the registers, the register transaction can occur in a different order.
```

Comment:
Can you add a note on the implications for the coprocessor? E.g. it probably needs buffering for all register operands depending on the number of outstanding accepted instructions or something like that.

Reply:
the coprocessor will need a buffer the id provided during issue, the mode, and either the instruction or information derived from it. Register operands do not have to be buffered, because they are provided later. Let's discuss

Reply:
If a coprocessor cannot apply back pressure and it needs to deal with out of order register transactions, then it either needs to provide buffering or limit the number of accepted transactions so that out of order register transactions are prevented.

Reply:
correct. it can even have a buffer size of 1. i this case, no new instruction will be accepted in issue transaction before the registers are supplied. Will elaborate on that.